### ISSUE TITLE
Error with entrypoint.py while creating an AMI

### ISSUE THREAD

**Issue by coderodyhpc** [aws/aws-parallelcluster#4146]
Hello,
The creation of an AMI with `pcluster build-image` is failing with the (long) log pointing to some problem with the  ImageSchemaValidator:
```
...
ImageSchemaValidator:
2022-06-27 21:28:32,755 - ERROR - entrypoint.py:250:main() - {"configurationValidationErrors": [{"level": "ERROR", "type": "ImageSchemaValidator", "message": "[('Build', {'ParentImage': ['String does not match expected pattern.']})]"}], "message": "Invalid image configuration."}
Traceback (most recent call last):
  File "/home/ubuntu/.local/lib/python3.10/site-packages/pcluster/models/imagebuilder.py", line 421, in _validate_config
    validation_failures = self.config.validate(validator_suppressors)
  File "/home/ubuntu/.local/lib/python3.10/site-packages/pcluster/models/imagebuilder.py", line 324, in config
    self.__config = ImageBuilderSchema().load(parse_config(self.__source_config_text))
  File "/home/ubuntu/.local/lib/python3.10/site-packages/marshmallow/schema.py", line 722, in load
    return self._do_load(
  File "/home/ubuntu/.local/lib/python3.10/site-packages/marshmallow/schema.py", line 909, in _do_load
    raise exc
marshmallow.exceptions.ValidationError: {'Build': {'ParentImage': ['String does not match expected pattern.']}}
...
```
It looks a bit odd because the original image is defined as `ParentImage: ami-070650c005cce4203` in the buid script. 
I'm not sure which is the root of the problem. Thanks. 

[pcluster-cli.log](https://github.com/aws/aws-parallelcluster/files/8996026/pcluster-cli.log)

**Comment by demartinofra** [comment]
Hi,

according to the shared logs the image configuration used when the error occurs is `ParentImage: GLADIUM`. 

```
2022-06-27 21:28:32,321 - INFO - entrypoint.py:227:run() - Parsed CLI arguments: args(Namespace(operation='build-image', suppress_validators=None, validation_failure_level=None, dryrun=None, rollback_on_failure=None, region='us-east-1', image_configuration='Build:\n  InstanceType: t4g.large\n  ParentImage: GLADIUM\n', image_id='GLADIUM', query=None, 
```

while in the next iteration happening at `2022-06-27 21:31:45,809` the ParentImage is correctly set to `ami-070650c005cce4203` and the error does not occurr.

**Comment by coderodyhpc** [comment]
Hi @**demartinofra**, 
Let me try again but, before doing so, a quick question. Maybe I misread the documentation but the command to create the AMI is:
```
pcluster build-image --image-id IMAGE_ID --image-configuration IMAGE_CONFIG.yaml --region REGION
```
The yaml file and region are straightforward but does the parameter IMAGE-ID refer to the base AMI ID or to the name of the new AMI?
Thanks.

**Comment by coderodyhpc** [comment]
Hi @**demartinofra**, 
I'm not really sure if I understand what you're referring to. I tried again using a new stack name (HASTA) but it's still failing. In the attached log, all the _ParentImage_ refer to _ami-070650c005cce4203_ as far as I can see. The stack also shows that it couldn't create the ParallelClusterImage.
Thanks.
P.S. The yaml file is simply:
```
Build:
  InstanceType: t4g.large
  ParentImage: ami-070650c005cce4203
```
[pcluster-cli.log](https://github.com/aws/aws-parallelcluster/files/9020996/pcluster-cli.log)
![STACK_HASTA](https://user-images.githubusercontent.com/79646720/176711172-3fc48368-d5c7-4a0a-95bd-3d7de31e55e0.jpg)

**Comment by coderodyhpc** [comment]
Files:

**Comment by demartinofra** [comment]
Hi @coderodyhpc 

ok, the error you are seeing here is unrelated to the error you pointed out in the initial message, where you were mentioning an error in the ImageSchemaValidator.

I believe the problem now is that you are trying to use an AMI for Ubuntu 22 that is not supported by ParallelCluster. You should be able to troubleshoot build failures by looking at the image build logs stored in cloudwatch. You can use the `get-image-log-events` and `export-image-logs` commands to get those logs. (see https://docs.aws.amazon.com/parallelcluster/latest/ug/building-custom-ami-v3.html)

**Comment by coderodyhpc** [comment]
Hi @**demartinofra**, 
I've followed the same procedure but on Monday I had to reinstall aws-parallelcluster a couple of times due to some minor hiccups; I thought that everything was fine but maybe there was something iffy causing the ParentImage confusion and failure in the AMI creation. Before actually running into that error, my concern was whether it would accept an Ubuntu22.04 base AMI. It seems that the answer to that question is 'no' and I'll have to wait until Ubuntu22 becomes available. Thanks.
